### PR TITLE
SAR-10775 Added support to Handoffs for language field

### PR DESCRIPTION
**Pull request by NathanFarnell**
Requires
https://github.com/hmrc/incorporation-frontend-stubs/pull/35

### CODE REVIEW

File: app/controllers/handoff/CorporationTaxDetailsController.scala [hunk 1]
```diff
@@ -63,7 +63,7 @@ class CorporationTaxDetailsController @Inject()(val authConnector: PlayAuthConne
       ctAuthorisedHandoff("HO2", requestData) {
         registeredHandOff("HO2", requestData) { _ =>
           handBackService.processCompanyDetailsHandBack(requestData).map {
-            case Success(_) => Redirect(controllers.reg.routes.PPOBController.show)
+            case payload@Success(_) => Redirect(controllers.reg.routes.PPOBController.show).withLang(Lang(payload.get.language))
```
Comment (MJCallahanPage) [inline app/controllers/handoff/CorporationTaxDetailsController.scala:66]: I think using pattern matching to extract the inner here would be better so you can avoid the `.get` later
```suggestion
            case Success(payload) => Redirect(controllers.reg.routes.PPOBController.show).withLang(Lang(payload.language))
```

File: app/controllers/handoff/CorporationTaxSummaryController.scala [hunk 2]
```diff
@@ -47,7 +48,7 @@ class CorporationTaxSummaryController @Inject()(val authConnector: PlayAuthConne
       ctAuthorisedHandoff("HO4", requestData) {
         registeredHandOff("HO4", requestData) { _ =>
           handBackService.processSummaryPage1HandBack(requestData).map {
-            case Success(_) => Redirect(controllers.reg.routes.SummaryController.show)
+            case payload@Success(_) => Redirect(controllers.reg.routes.SummaryController.show).withLang(Lang(payload.get.language))
```
Comment (MJCallahanPage) [inline app/controllers/handoff/CorporationTaxSummaryController.scala:51]: Ditton:
```suggestion
            case Success(payload) => Redirect(controllers.reg.routes.SummaryController.show).withLang(Lang(payload.language))
```

File: app/controllers/handoff/GroupController.scala [hunk 3]
```diff
@@ -54,11 +56,11 @@ class GroupController @Inject()(val authConnector: PlayAuthConnector,
       ctAuthorisedHandoff("HO3-1", requestData) {
         registeredHandOff("HO3-1", requestData) { regID =>
           handBackService.processGroupsHandBack(requestData).flatMap {
-            case Success(GroupHandBackModel(_, _, _, _, _, Some(true))) =>
-              pscHandOffToGroupsIfDataInTxApi(regID)
-            case Success(GroupHandBackModel(_, _, _, _, _, Some(false))) =>
+            case payload@Success(GroupHandBackModel(_, _, _, _, _, _ , Some(true))) =>
+              pscHandOffToGroupsIfDataInTxApi(regID, payload.get.language)
+            case payload@Success(GroupHandBackModel(_, _, _, _, _, _ , Some(false))) =>
```
Comment (MJCallahanPage) [inline app/controllers/handoff/GroupController.scala:61]:
```suggestion
            case Success(GroupHandBackModel(_, _, _, _, lang, _ , Some(true))) =>
              pscHandOffToGroupsIfDataInTxApi(regID, lang)
            case Success(GroupHandBackModel(_, _, _, _, lang, _ , Some(false))) =>
```

File: app/controllers/handoff/GroupController.scala [hunk 4]
```diff
@@ -54,11 +56,11 @@ class GroupController @Inject()(val authConnector: PlayAuthConnector,
...
               groupService.dropGroups(regID).map { _ =>
-                Redirect(controllers.handoff.routes.GroupController.PSCGroupHandOff)
+                Redirect(controllers.handoff.routes.GroupController.PSCGroupHandOff).withLang(Lang(payload.get.language))
```
Comment (MJCallahanPage) [inline app/controllers/handoff/GroupController.scala:63]:
```suggestion
                Redirect(controllers.handoff.routes.GroupController.PSCGroupHandOff).withLang(Lang(lang))
```

File: app/controllers/handoff/RegistrationConfirmationController.scala [hunk 5]
```diff
@@ -52,11 +53,11 @@ class RegistrationConfirmationController @Inject()(val authConnector: PlayAuthCo
         registered {
           regid =>
             handBackService.decryptConfirmationHandback(requestData) flatMap {
-              case Success(s) => handBackService.storeConfirmationHandOff(s, regid).flatMap {
+              case payload@Success(s) => handBackService.storeConfirmationHandOff(s, regid).flatMap {
```
Comment (MJCallahanPage) [inline app/controllers/handoff/RegistrationConfirmationController.scala:56]:
```suggestion
              case Success(s) => handBackService.storeConfirmationHandOff(s, regid).flatMap {
```

File: app/controllers/handoff/RegistrationConfirmationController.scala [hunk 6]
```diff
@@ -52,11 +53,11 @@ class RegistrationConfirmationController @Inject()(val authConnector: PlayAuthCo
...
                 case _ if handBackService.payloadHasForwardLinkAndNoPaymentRefs(s) => getPaymentHandoffResult(externalID)
-                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show))
-                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show))
-                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage))
+                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(payload.get.language)))
+                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(payload.get.language)))
+                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage).withLang(Lang(payload.get.language)))
```
Comment (MJCallahanPage) [inline app/controllers/handoff/RegistrationConfirmationController.scala:60]:
```suggestion
                case ConfirmationReferencesSuccessResponse(_) => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(s.language)))
                case DESFailureRetriable => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.show).withLang(Lang(s.language)))
                case _ => Future.successful(Redirect(controllers.reg.routes.ConfirmationController.deskproPage).withLang(Lang(s.language)))
```

Comment (MJCallahanPage) [reply app/controllers/handoff/RegistrationConfirmationController.scala:60]: You could rename the inner extraction on the pattern matching to be `payload` if you wanted it to be consistent and more understandable than what was there originally as `s` :)

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
Looks good, just some minor refactor suggestions to make use of the inner extraction on the pattern matching and therefore remove the `.get`

**Review by MJCallahanPage — APPROVED** [review]
🟢 lgtm - will just wait for the PR builder then run ATs locally.

**Comment by Sellie96** [comment]
> 🟢 lgtm - will just wait for the PR builder and run ATs locally.

Don't forget the IFS branch if you're running the AT's :)

**Pull request merged**